api/src/main/java/com/spotify/ffwd/model/Metric.java

```diff
@@ -48,4 +54,28 @@
     public Batch.Point toBatchPoint() {
         return new Batch.Point(key, tags, resource, value, time.getTime());
     }
+
+    public String generateHash() {
+        final Hasher hasher = HASH_FUNCTION.newHasher();
+
+        if (key != null) {
+            hasher.putString(key, Charsets.UTF_8);
+        }
+
+        for (final Map.Entry<String, String> kv : tags.entrySet()) {
```

Comment:
Is the order for entrySet guaranteed to be the same every time its called? I know it will for HashMaps, but I'm not sure about other Map implementations.

Reply:
Good question. Looking at the `ProtobufDecoder` and the `JsonObjectMapperDecoder` the Metric is constructed with a `HashMap<>`. 

Is this a case where it would be better to be specific about the map implemented required or sort the keys before iterating over them?

Reply:
I think so, might be good to change the type for `tags` to a HashMap if we can't confirm the general case for Map.

Reply:
something got messed up. adding the `TreeMap` back.